src/Microsoft.AspNet.Mvc.DataAnnotations/DependencyInjection/MvcDataAnnotationsMvcCoreBuilderExtensions.cs

```diff
@@ -71,6 +73,8 @@ internal static void AddDataAnnotationsServices(IServiceCollection services)
         {
             services.TryAddEnumerable(
                 ServiceDescriptor.Transient<IConfigureOptions<MvcOptions>, MvcDataAnnotationsMvcOptionsSetup>());
+            var attributeAdapterProvider = new ValidationAttributeAdapterProvider();
+            services.TryAddSingleton<IValidationAttributeAdapterProvider>(s => attributeAdapterProvider);
```

Comment:
Why isn't this just `TryAddSingleton<IValidationAttributeAdapterProvider, ValidationAttributeAdapterProvider>()`?